app/src/main/java/com/eveningoutpost/dexdrip/webservices/WebServiceSgv.java

```diff
@@ -106,8 +112,48 @@ public WebResponse request(String query) {
         // whether to include data which doesn't match the current sensor
         final boolean ignore_sensor = Home.get_follower() || cgi.containsKey("all_data");
 
-        final List<BgReading> readings = BgReading.latest(count, ignore_sensor);
+
+        // Store a cache of the last BgReading.latest() query for the duration in which there is no
+        // new latest reading. Since obtaining the latest reading is fast, but a larger number of
+        // readings is significantly slower, this optimizes the most often use case.
+        List<BgReading> bgr = BgReading.latest(1, ignore_sensor);
+        BgReading latestReading = null;
+        if (bgr != null) {
+            Iterator<BgReading> it = bgr.iterator();
+            latestReading = it.hasNext() ? it.next() : null;
+        }
+
+        List<BgReading> readings;
+        if (this.cachedReadings != null && latestReading != null &&
+            this.cachedReadings.size() > 0 && latestReading.uuid.equals(this.cachedReadings.iterator().next().uuid) &&
+            count <= this.cachedReadings.size())
+        {
+            if (count == this.cachedReadings.size()) {
+                UserError.Log.d(TAG, "Using cached readings");
+                readings = this.cachedReadings;
+            } else {
+                UserError.Log.d(TAG, "Copying "+count+" of "+this.cachedReadings.size()+" cached readings");
+                readings = new ArrayList<>();
+                Iterator<BgReading> it = this.cachedReadings.iterator();
+                for (int i=0; i<count; i++) {
+                    readings.add(it.next());
+                }
+            }
+        } else {
+            UserError.Log.d(TAG, "Fetching latest "+count+" readings from BgReading");
+            readings = BgReading.latest(count, ignore_sensor);
+            this.cachedReadings = readings;
+        }
+
+        // Prevent the cache from growing too large. 1000 entries is about 3.5 days
+        if (cachedJson.size() > 1000) {
```

Comment:
Isn't it more efficient to use a `LinkedHashMap` and implement `removeEldestEntry` in such a way that the [oldest entries are removed automatically](https://docs.oracle.com/javase/8/docs/api/java/util/LinkedHashMap.html#removeEldestEntry-java.util.Map.Entry-), if the map contains 1.000 entries already and a new element gets added, instead of removing all entries?

Reply:
Clever solution -- done.